src/pages/Trade/Swap/index.tsx

```diff
@@ -432,6 +440,16 @@ function SwapPage() {
     }
   }, [form, preselectedAsset2Address, setPreselectedAsset2Address]);
 
+  useEffect(() => {
```

Comment:
initializing the amount of an undefined asset doesn't seems to be related to the reported case.
Also it causes another issue the amount becomes empty when you enter an amount first and then selects another asset, which is not expected.

Reply:
fixed it on the last commit: [443c1c9](https://github.com/dezswap/dezswap-web-app/pull/231/commits/443c1c90898888e059292409a829aacdd1f4adb3)

it keeps amount on the unselected asset even when the opposite changes:

https://github.com/user-attachments/assets/57701d8c-45c3-4831-91c3-90fa9faeec9c

and resolves the reported issues:

https://github.com/user-attachments/assets/cb891ddd-e725-4663-840a-6c9770a13b51